### PR TITLE
Add caption support for datasources

### PR DESCRIPTION
**Pull request by RussTheAerialist**
Addresses #98

### CODE REVIEW

File: tableaudocumentapi/datasource.py [hunk 1]
```diff
@@ -207,6 +208,15 @@ def name(self):
     def version(self):
         return self._version
 
+    @property
+    def caption(self):
+        return self._caption
+
+    @caption.setter
+    def caption(self, value):
+        self._datasourceXML.attrib['caption'] = value
```
Comment (t8y8) [inline tableaudocumentapi/datasource.py]: Nit: We do `.set` everywhere else (or everywhere in `connections`)

File: test/test_datasource.py [hunk 2]
```diff
@@ -22,6 +26,19 @@ class DataSourceFieldsTDS(unittest.TestCase):
...
+        self.to_delete = set()
+
+    def cleanUp(self):
+        for path in self.to_delete:
+            if os.path.isdir(path):
+                shutil.rmtree(path, ignore_errors=True)
+            elif os.path.isfile(path):
+                os.unlink(path)
+
+    def get_temp_file(self, filename):
```
Comment (t8y8) [inline test/test_datasource.py]: We have 3 ways of doing temp file management now.

`xfile.temporary_directory` has a context manager that creates and deletes a temp dir
`bvt.py` does a less advanced version of your cleanup methods here
Then these cleanup methods here.

This can be another PR, maybe we want to unify them, at least across the tests?

File: tableaudocumentapi/datasource.py [hunk 3]
```diff
@@ -207,6 +208,15 @@ def name(self):
     def version(self):
         return self._version
 
+    @property
+    def caption(self):
+        return self._caption
+
+    @caption.setter
```
Comment (t8y8) [inline tableaudocumentapi/datasource.py]: Are all datasources guaranteed to have a caption or would you ever want to remove it?

We could apply the `None` strategy like we do for port

Comment (RussTheAerialist) [inline tableaudocumentapi/datasource.py]: I think it's more idiomatic to do `del ds.caption`

**Comment by t8y8** [comment]
:rocket:
